docs/paths.adoc

```diff
@@ -0,0 +1,148 @@
+
+== Paths
+
+A `path` is a way to access a field or register. It is a compile-time list of
+strings.
+
+=== Construction and manipulation
+
+Paths can be constructed using UDLs which are in the `groov::literals` namespace.
+
+[source,cpp]
+----
+using namespace groov::literals;
+
+auto p1 = "reg"_r;
+auto p2 = "field"_f;
+auto p3 = "group"_g;
```

Comment:
question: what is `_g` used for? i don't see how it is used with the current api.

Reply:
I don't think it is used. I've removed it from the docs.